### PR TITLE
Backport 26326 to make our Sage work with NetworkX 2.2

### PR DESCRIPTION
**Pull request by saraedum**

### CODE REVIEW

**Comment by saraedum** [comment]
This does not sound like something that Sage is responsible for @isuruf:
```
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/sagelib_1539199726819/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages/numpy/core/__init__.py", line 16, in <module>
    from . import multiarray
ImportError: libpython3.6m.so.1.0: cannot open shared object file: No such file or directory
```

I'll rerun the test, maybe it'll go away on its own.

**Comment by isuruf** [comment]
It's not. Broken python packages were pulled.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -88,6 +90,8 @@ requirements:
     - mpc 1.1.*
     - mpfi 1.5.*
     - mpfr
+    # Since we apply 26326, we can only work with NetworkX 2.2
+    - networkx 2.2
```
Comment (isuruf) [inline recipe/meta.yaml]: This is needed in run right?

Comment (saraedum) [reply recipe/meta.yaml]: Oops, you're right.

Comment (isuruf) [reply recipe/meta.yaml]: Instead of run, add it to `run_constrained`. Let me find an example for you

**Comment by saraedum** [comment]
Nice, I didn't know about this :)

**Comment by saraedum** [comment]
I think I have misread the upstream ticket. I got the impression that the random problem had been sorted out already. So now we've got some new errors with random graphs. Maybe we should revert this and instead require NetworkX 2.1. Sorry for the noise.

**Pull request merged**